consensus/hotstuff/verification/combined_verifier_v2.go

```diff
@@ -0,0 +1,142 @@
+// +build relic
+
+package verification
+
+import (
+	"fmt"
+
+	"github.com/onflow/flow-go/consensus/hotstuff"
+	"github.com/onflow/flow-go/consensus/hotstuff/model"
+	"github.com/onflow/flow-go/consensus/hotstuff/signature"
+	"github.com/onflow/flow-go/crypto"
+	"github.com/onflow/flow-go/crypto/hash"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+	modulesig "github.com/onflow/flow-go/module/signature"
+)
+
+// CombinedVerifier is a verifier capable of verifying two signatures for each
+// verifying operation. The first type is a signature from an aggregating signer,
+// which verifies either the single or the aggregated signature. The second type is
+// a signature from a threshold signer, which verifies either the signature share or
+// the reconstructed threshold signature.
+type CombinedVerifierV2 struct {
+	committee      hotstuff.Committee
+	staking        hash.Hasher
+	beacon         hash.Hasher
+	keysAggregator *stakingKeysAggregator
+	merger         module.Merger
+}
+
+// NewCombinedVerifier creates a new combined verifier with the given dependencies.
+// - the hotstuff committee's state is used to retrieve the public keys for the staking signature;
+// - the staking tag is used to create hasher to verify staking signatures;
+// - the beacon tag is used to create hasher to verify random beacon signatures;
+// - the merger is used to combined & split staking & random beacon signatures;
+func NewCombinedVerifierV2(committee hotstuff.Committee, stakingTag string, beaconTag string, merger module.Merger) *CombinedVerifierV2 {
+	return &CombinedVerifierV2{
+		committee:      committee,
+		staking:        crypto.NewBLSKMAC(stakingTag),
+		beacon:         crypto.NewBLSKMAC(beaconTag),
```

Comment:
I suggest to add more clarity
```suggestion
		stakingHasher:        crypto.NewBLSKMAC(stakingTag),
		beaconHasher:         crypto.NewBLSKMAC(beaconTag),
```